### ISSUE TITLE
IcDialog: Display issues at >200% Zoom

### ISSUE THREAD

**Issue by GCHQ-Developer-299** [mi6/ic-ui-kit#2868]
## Summary of the bug 
Reported by an internal user - when browser zoom is above 200% (and crucially at the 400% specified by [WCAG Criterion 1.4.10](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html)) IcDialogs display incorrectly.

## 🪜 How to reproduce 
Issue can be seen on [Guidance Site dialog examples](https://design.sis.gov.uk/components/dialog/code#sizes) and [Dialog Storybook ](https://mi6.github.io/ic-ui-kit/branches/develop/web-components/iframe.html?args=&id=web-components-dialog--sizes&viewMode=story)

Once dialog is open, zoom the page above 200% and see that it spills above the viewport. The dialog title is not visible. 

## 📸 Screenshots or code 
<img width="1445" alt="Screenshot 2024-12-05 at 11 12 32" src="https://github.com/user-attachments/assets/edd4f2fc-5dba-4287-93b6-f4e0cc66cdf4">

## 🧐 Expected behaviour 
For the full dialog to be visible (potentially through internal scrolling?)
